### ISSUE TITLE
_add_scale_1d2d3d

### ISSUE THREAD

**Issue by ryandok** [NVIDIA-AI-IOT/torch2trt#664]
Great work! Thank you.
It is there problem with the function "_add_scale_1d2d3d".
When I use InstancNorm with "afiine=True" or "track_running_stats=True", the original output doesn't match to trt_output.

They both use this function.

    if not use_input_stats:
        
        # equivalent to batch norm
        scale = 1.0 / np.sqrt(running_var.detach().cpu().numpy() + eps)
        offset = -running_mean.detach().cpu().numpy() * scale
        power = np.ones_like(scale)
        
        if weight is not None:
            scale *= weight.detach().cpu().numpy()
            offset += bias.detach().cpu().numpy()
            
        result_trt = _add_scale_1d2d3d(ctx.network, input._trt, trt.ScaleMode.CHANNEL, offset, scale, power) ****here!!****

Please help TAT

**Comment by ryandok** [comment]
Hey, maybe I found the problem.
When I used InstanceNorm2D, the original output match to trt_output.
But 3D not.
The difference is below:
************************************************************************
def _add_scale_1d2d3d(network, x_trt, mode, offset, scale, power):
    ndim = len(x_trt.shape)
    
    y_trt = x_trt
    
    # shape to 2D      #here!!!!!!!!!!!!!!!!
    if ndim != 3:
        layer = network.add_shuffle(y_trt)
        layer.reshape_dims = (x_trt.shape[0], x_trt.shape[1], -1)  # NCH -> NCHW
        y_trt = layer.get_output(0)
        
    y_trt = network.add_scale(y_trt, mode, offset, scale, power).get_output(0)

    # shape to original dimension
    if ndim != 3:     #here!!!!!!!!!!!!!!!!
        layer = network.add_shuffle(layer.get_output(0))
        layer.reshape_dims = tuple(x_trt.shape)
        y_trt = layer.get_output(0)
    
    return y_trt
************************************************************************
I think the problem is shape to 2D when the dim is 3D

**Comment by ryandok** [comment]
Hi @jaybdub 

Thanks for your answer.

But still not yet.
Sorry.

Best,
Ryan
